core/src/logging/filter/filter_builders.rs

```diff
@@ -0,0 +1,73 @@
+use crate::log_id::LogLevel;
+
+use super::{AddonFilter, FilterConfig, LogIdAddonFilter, LogIdModuleFilter};
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct FilterConfigBuilder {
+    // TODO: negation is confusing, use affirmation instead (no_general_logging -> general_logging)
+    no_general_logging: bool,
+    general_level: LogLevel,
+    general_addons: Vec<AddonFilter>,
+    /// LogIds set with `on[LogId]`
+    allowed_global_ids: Vec<LogIdAddonFilter>,
+    allowed_modules: Vec<LogIdModuleFilter>,
+}
+
+impl FilterConfigBuilder {
+    pub fn new(level: LogLevel) -> Self {
+        Self {
+            general_level: level,
+            ..Default::default()
+        }
+    }
+
+    pub fn level(mut self, level: LogLevel) -> Self {
```

Comment:
why would you need this function?
`new()` already takes a level, so this function would just overwrite the initial one.

Reply:
That is a very valid critique. So, right now I have the builder as a public struct. The idea being, users can use the builder directly to create a filter (instead of doing it with a string). Ideally, we would also create helper types so that other building blocks of the filter are easier to construct (and pass to builder) as well. 

With that being said, what should the ideal API look like. Do we want to set a default log level, or should we always force users to pass a log level? 

If we want default, then `::builder()` should produce a builder with default values. If not, then users should always pass a level to the `::builder()` function. 

I am kind of leaning towards explicitness (no default values), because as soon as users (developers) want to customise anything in a filter, they probably also want a custom log level. This also makes it a little more intuitive to use, since people don't need to read documentation to find out what the default is etc.

Reply:
I agree with you on explicitness.
If you use the `builder()` function, you want to set values different to the default values of the logid filter struct.
But some default value must be set for the filter struct, because these values are taken if no `set_filter()` call is used.

Reply:
Changed this in new commits.